rest_framework/fields.py

```diff
@@ -1376,7 +1384,22 @@ def to_internal_value(self, value):
         self.fail('invalid', format='[DD] [HH:[MM:]]ss[.uuuuuu]')
 
     def to_representation(self, value):
-        return duration_string(value)
+        output_format = getattr(self, 'format', api_settings.DURATION_FORMAT)
+
+        if output_format is None:
+            return value
+
+        if isinstance(output_format, str):
+            if output_format.lower() == ISO_8601:
+                return duration_iso_string(value)
+
+            if output_format.lower() == DJANGO_DURATION_FORMAT:
+                return duration_string(value)
+
+        raise ValueError(
```

Comment:
I think I would have made that a `RuntimeError`, but I suppose that's highly subjective :)